lifecycle-operator/controllers/lifecycle/keptnworkloadversion/controller.go

```diff
@@ -121,26 +120,15 @@ func (r *KeptnWorkloadVersionReconciler) Reconcile(ctx context.Context, req ctrl
 		spanWorkloadTrace.AddEvent("WorkloadVersion Pre-Deployment Tasks started", trace.WithTimestamp(time.Now()))
 	}
 
-	if !workloadVersion.IsPreDeploymentSucceeded() {
-		reconcilePre := func(phaseCtx context.Context) (apicommon.KeptnState, error) {
-			return r.reconcilePrePostDeployment(ctx, phaseCtx, workloadVersion, apicommon.PreDeploymentCheckType)
-		}
-		result, err := phaseHandler.HandlePhase(ctx, ctxWorkloadTrace, r.getTracer(), workloadVersion, phase, reconcilePre)
-		if !result.Continue {
-			return result.Result, err
-		}
+	result, err := r.doPreDeploymentTaskPhase(ctx, workloadVersion, phaseHandler, ctxWorkloadTrace)
+	if !result.Continue {
```

Comment:
since we re-use result and err for all steps, maybe it is worth scoping them to each if, although it makes the line longer 🤔 
```suggestion
	if result, err := r.doPreDeploymentTaskPhase(ctx, workloadVersion, phaseHandler, ctxWorkloadTrace); !result.Continue {
```